src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -1927,6 +1938,20 @@ void Cleanup()
                 CurrentReader = null;
             }
 
+            if (CurrentCopyOperation != null)
+            {
+                try
+                {
+                    // Will never complete asynchronously (stream is already closed)
+                    CurrentCopyOperation.Dispose();
```

Comment:
Maybe we should call DisposeAsync and assert on the resulting task that it completed synchronously etc., for sanity (for reader above as well).